taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/ServerProjectResource.java

```diff
@@ -0,0 +1,34 @@
+package com.digirati.taxman.rest.server;
+
+import com.digirati.taxman.common.taxonomy.ProjectModel;
+import com.digirati.taxman.rest.server.taxonomy.ProjectModelRepository;
+import com.digirati.taxman.rest.taxonomy.ProjectPath;
+import com.digirati.taxman.rest.taxonomy.ProjectResource;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.core.Response;
+
+@ApplicationScoped
+public class ServerProjectResource implements ProjectResource {
+
+    @Inject
+    ProjectModelRepository projectModelRepository;
+
+    @Override
+    public Response createProject(@Valid ProjectModel project) {
+        return Response.ok(projectModelRepository.create(project)).build();
+    }
+
+    @Override
+    public Response getProject(ProjectPath projectPath) {
+        return Response.ok(projectModelRepository.find(projectPath.getProjectSlug())).build();
+    }
+
+    @Override
+    public Response updateProject(ProjectPath projectPath, @Valid ProjectModel project) {
+        projectModelRepository.update(project);
+        return Response.noContent().build();
```

Comment:
I'm concerned that `204 No Content` is not the correct response to return here. Typically I've seen this used in response to a `DELETE`, where the target resource has been deleted and thus there's nothing to return.

For any kind of update call (e.g. `POST`, `PUT`) I think it makes sense to return _the updated object_ a response to that operation, which would imply `200 OK`. This allows for the client to immediately update their view of that resource (e.g. in the browser) without having to make additional calls.

I understand that there may be other similar calls outside of the project operations where we're returning `204 No Content` in response to an update operation, so this may be a larger issue.

Reply:
This is the case for the updates on concepts and concept schemes as well so I'd prefer to tackle them all in one go. I'm not sure I agree that returning the modified object would be useful to the client though, since in all three cases we're just storing what they told us to store rather than doing any server side processing on that data